hera_cal/tests/test_frf.py

```diff
@@ -0,0 +1,97 @@
+import nose.tools as nt
+import os
+import shutil
+import numpy as np
+import sys
+from pyuvdata import UVData
+from pyuvdata import utils as uvutils
+import hera_cal as hc
+from hera_cal.data import DATA_PATH
+from collections import OrderedDict as odict
+import copy
+import glob
+
+
+def test_timeavg_waterfall():
+    fname = os.path.join(DATA_PATH, "zen.2458042.12552.xx.HH.uvXA")
+
+    uvd = UVData()
+    uvd.read_miriad(fname)
+    
+    d = uvd.get_data(24, 25)
+    f = uvd.get_flags(24, 25)
+    n = uvd.get_nsamples(24, 25)
+    t = np.unique(uvd.time_array)
+    fr = uvd.freq_array.squeeze()
+    l = []
+    for _l in uvd.lst_array:
+        if _l not in l:
+            l.append(_l)
+    l = np.array(l)
+    antpos, ants = uvd.get_ENU_antpos()
+    blv = antpos[ants.tolist().index(24)] - antpos[ants.tolist().index(25)]
+
+    # test basic execution
+    ad, af, an, al, aea = hc.frf.timeavg_waterfall(d, 25, verbose=False)
+    nt.assert_equal(ad.shape, (3, 64))
+    nt.assert_equal(af.shape, (3, 64))
+    nt.assert_equal(an.shape, (3, 64))
+    nt.assert_false(np.any(af))
+    nt.assert_almost_equal(an[1, 0], 25.0)
+    nt.assert_almost_equal(an[2, 0], 10.0)
+
+    # test rephase
+    ad, af, an, al, aea = hc.frf.timeavg_waterfall(d, 25, flags=f, rephase=True, lsts=l, freqs=fr, bl_vec=blv, 
+                                                    nsamples=n, extra_arrays=dict(times=t), verbose=False)
+    nt.assert_equal(ad.shape, (3, 64))
+    nt.assert_equal(af.shape, (3, 64))
+    nt.assert_equal(an.shape, (3, 64))
+    nt.assert_true(np.any(af))
+    nt.assert_equal(len(al), 3)
+    nt.assert_equal(len(aea['avg_times']), 3)
+    nt.assert_almost_equal(an.max(), 25.0)
+
+
+class Test_FRFilter:
+
+    def setUp(self):
+        self.fname = os.path.join(DATA_PATH, "zen.2458042.12552.xx.HH.uvXA")
+        self.F = hc.frf.FRFilter()
+        self.uvd = UVData()
+        self.uvd.read_miriad(self.fname)
+
+    def test_load_data(self):
+        self.F.load_data(self.fname)
+        nt.assert_equal(self.F.inp_uvdata, self.uvd)
+        self.F.load_data(self.uvd)
+        nt.assert_equal(self.F.inp_uvdata, self.uvd)
+
+    def test_timeavg_data(self):
+        self.F.load_data(self.uvd)
+        self.F.timeavg_data(35, rephase=True)
+        nt.assert_equal(self.F.Navg, 3)
+
+        self.F.timeavg_data(1e10, rephase=True, verbose=False)
+        nt.assert_equal(self.F.Navg, 60)
+
+        # exceptions
+        nt.assert_raises(AssertionError, self.F.timeavg_data, 1.0)
+
+    def test_write_data(self):
+        self.F.load_data(self.uvd)
+        self.F.timeavg_data(35, rephase=False, verbose=False)
+        u = self.F.write_data("./out.uv", write_avg=True, filetype='miriad', overwrite=True)
+        nt.assert_true(os.path.exists("./out.uv"))
+        uv = UVData()
+        uv.read_miriad('./out.uv')
+        nt.assert_equal(u, uv)
+
+        u = self.F.write_data("./out.uv", overwrite=False)
+        nt.assert_equal(u, None)
```

Comment:
I'm confused about this test. What is it testing for and why does it work?

Reply:
this test is making sure that the `overwrite` kwarg works. If `out.uv` already exists (which it does) and `overwrite=False`, the method **does not** create a `new_uvd` and returns a None, which the assert is checking.